sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageSourceBase.java

```diff
@@ -146,7 +159,18 @@
       return ImmutableList.of();
     }
 
-    Schema sessionSchema = new Schema.Parser().parse(readSession.getAvroSchema().getSchema());
+    Schema sessionSchema;
+    if (readSession.getDataFormat() == DataFormat.ARROW) {
+      org.apache.arrow.vector.types.pojo.Schema schema =
+          ArrowConversion.arrowSchemaFromInput(
+              readSession.getArrowSchema().getSerializedSchema().newInput());
+      org.apache.beam.sdk.schemas.Schema beamSchema =
+          ArrowConversion.ArrowSchemaTranslator.toBeamSchema(schema);
+      sessionSchema = AvroUtils.toAvroSchema(beamSchema);
+    } else {
```

Comment:
explicitly check for `DataFormat.AVRO` here. The `else` case should raise an`IllegalArgumentException`. 

It's a little bit ugly that we have some of the format-specific logic right here, while the rest of it is in the `BigQueryStorage{Arrow,Avro}Reader`, but this is fine for now.

Reply:
Done